app/services/proofing_document_capture_session_result.rb

```diff
@@ -0,0 +1,59 @@
+class ProofingDocumentCaptureSessionResult
+  REDIS_KEY_PREFIX = 'dcs-proofing:result'.freeze
```

Comment:
This is nearly a full copy of `DocumentCaptureSessionResult` --- now that we have 2 of these, would it be worth having something that's like a `RedisJsonStruct` that we can make and then separate the value objects from the storage? I'm thinking like:

```rb
ProofingDocumentCaptureSessionResult = RedisJsonStruct.new(
  attrs: [:id, :pii, :result],
  redis_key_prefix: 'dcs-proofing:result'
)
# alternative, possibly easier to read
ProofingDocumentCaptureSessionResult = Struct.new(:id, :pii, :result, keyword_init: true) do
  def self.redis_key_prefix
    'dcs-proofing:result'
  end
end
```

Then we could have a separate like `RedisStructStorage` that does something like:

```rb
result = ProofingDocumentCaptureSessionResult.new(id: SecureRandom.hex, pii: pii, result: result)
RedisStructStorage.new.store(result)

deserialized_result = RedisStructStorage.new.load(result.id, type: ProofingDocumentCaptureSessionResult)

Reply:
The abstraction of the similar functionality was something I briefly discussed with @jmhooper so he may have thoughts here too.

My gut is to invoke the rule of threes since there may be more use cases along these lines. I don't know if I should be thinking of the common functionality more in terms of "private temporary document storage" or "Redis storage" right now.

I do think it would be good to align the methods and structure for the two classes a bit more though to move in that direction.

Reply:
I feel somewhat strongly about this, so I'm going to make a PR on to this branch with refactoring ideas

Reply:
PR: https://github.com/18F/identity-idp/pull/4217